### PR TITLE
this commit will clarify the structure of schemas as per feedback

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description

<what this pull request is doing>
Further cleaning up object models according to feedback from previous pull request
<what will be done in later PRs and not included here>

## Milestones
<the milestones or stories/features that this works towards>
Solidifying the object models

## Resources
<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>

### CODE REVIEW

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -23,37 +23,59 @@ model User {
...
-  userId         Int // if ingredient is an ingredient a user owns
-  user           User         @relation(fields: [userId], references: [id])
-  groceryListId  Int?
-  groceryList    GroceryList? @relation(fields: [groceryListId], references: [id])
+  id                   Int          @id @default(autoincrement())
+  ingredientName       String
+  quantity             Int
+  unit                 String
+  estimatedCostPerUnit Decimal
+  aisle                String?
```
Comment (paulshaoyuqiao) [inline backend/prisma/schema.prisma:31]: could you clarify what this means? Different store can have different aisles right (maybe not needed)

File: backend/prisma/schema.prisma [hunk 2]
```diff
@@ -23,37 +23,59 @@ model User {
...
+  productImage         String?
+  expiration           DateTime?
+  estimatedCost        Decimal?
+  userId               Int // if ingredient is an ingredient a user owns
```
Comment (paulshaoyuqiao) [inline backend/prisma/schema.prisma:35]: If each ingredient row has a user id, the table can get quite large because for common ingredients, almost every recipe will have them, and if we further multiply that by the number of users, it could become unscalable. same as the user field below.

**Review by paulshaoyuqiao — APPROVED** [review]
still feel like userId in ingredients table doesn't quite make sense, but will leave it up to you.

File: frontend/src/contexts/UserContext.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// import { createContext, useState, useContext } from "react";
```
Comment (paulshaoyuqiao) [inline frontend/src/contexts/UserContext.tsx:1]: can we just delete this file?

File: frontend/src/main.tsx [hunk 4]
```diff
@@ -2,10 +2,13 @@ import { StrictMode } from 'react'
 import { createRoot } from 'react-dom/client'
 import './index.css'
 import App from './App.tsx'
+// import { UserProvider } from './contexts/UserContext.jsx';
```
Comment (paulshaoyuqiao) [inline frontend/src/main.tsx:5]: same here

File: frontend/src/pages/LoginPage.tsx [hunk 5]
```diff
@@ -1,9 +1,58 @@
...
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, value } = event.target;
+        setFormData(prev => ({ ...prev, [name]: value }));
+    }
+
+    function handleSubmit(event: React.FormEvent) {
+        event.preventDefault();
+        // console.log("Signup form submitted!", formData);
+        // check password is valid length
+        if (!formData.username || !formData.password) {
```
Comment (paulshaoyuqiao) [inline frontend/src/pages/LoginPage.tsx:22]: might be good to refactor this into a separate helper method that validates whether everything in the formData is valid.

File: frontend/src/pages/LoginPage.tsx [hunk 6]
```diff
@@ -1,9 +1,58 @@
...
+    const [formData, setFormData] = useState({ username: "", password: "" });
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, value } = event.target;
+        setFormData(prev => ({ ...prev, [name]: value }));
+    }
+
+    function handleSubmit(event: React.FormEvent) {
+        event.preventDefault();
+        // console.log("Signup form submitted!", formData);
```
Comment (paulshaoyuqiao) [inline frontend/src/pages/LoginPage.tsx:20]: delete

File: frontend/src/pages/LoginPage.tsx [hunk 7]
```diff
@@ -1,9 +1,58 @@
...
+        // check password is valid length
+        if (!formData.username || !formData.password) {
+            throw new Error("Username and password are required")
```
Comment (paulshaoyuqiao) [inline frontend/src/pages/LoginPage.tsx:23]: users won't be able to see this? You might want to instead show an error text element inline?

File: frontend/src/pages/LoginPage.tsx [hunk 8]
```diff
@@ -1,9 +1,58 @@
...
+        .then((userCredential) => {
+            // Signed in 
+            const user = userCredential.user;
+            console.log("signed in user ", user)
+            // ...
+        })
+        .catch((error) => {
+            const errorCode = error.code;
+            const errorMessage = error.message;
+            console.log("error ", errorCode, errorMessage)
```
Comment (paulshaoyuqiao) [inline frontend/src/pages/LoginPage.tsx:36]: do we want to handle the authentication error here and show users something didn't work?

File: frontend/src/pages/SignupForm.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        const { name, value } = event.target;
+        setFormData(prev => ({ ...prev, [name]: value }));
+    }
+
+    function handleSubmit(event: React.FormEvent) {
+        event.preventDefault();
+        // console.log("Signup form submitted!", formData);
+        // check password is valid length
+        if (!formData.username || !formData.password) {
+            throw new Error("Username and password are required")
```
Comment (paulshaoyuqiao) [inline frontend/src/pages/SignupForm.tsx:22]: have you tested this btw? do you have any video recording?

**Comment by Brianna-Flo** [comment]
> still feel like userId in ingredients table doesn't quite make sense, but will leave it up to you.

I was unsure of how I might go about this since the ingredient would have user specific information such as expiration date and quantity (example: I have a different expiration date of milk than another user). I have looked into other options such as storing the data as a JSON file but was wondering if you had any other ideas?

File: backend/index.ts [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+import { Request, Response } from  "express";
+const express = require('express');
+const app = express();
+const PORT = 3000;
+const cors = require('cors')
+app.use(cors())
+app.use(express.json())
+
+
+// import prisma
```
Comment (radhikadhomse) [inline backend/index.ts:11]: omit

File: backend/index.ts [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    res.json(newUser)
+})
+
+
+app.listen(PORT, () => {
+  console.log(`Server is running on http://localhost:${PORT}`)
+})
+
+app.get('/', (req: Request, res: Response) => {
+  res.send('Welcome to my app!')
```
Comment (radhikadhomse) [inline backend/index.ts:34]: lets put a todo to change this

File: backend/index.ts [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+    const { firebaseId, email } = req.body
+    const newUser = await prisma.user.create({
+        data: {
+            firebaseId,
+            email,
+        }
+    })
+    console.log(newUser)
+    res.json(newUser)
+})
```
Comment (radhikadhomse) [inline backend/index.ts:26]: in a future PR, you could add a try / catch around the user creation

File: backend/prisma/migrations/20250625222403_added_initial_models/migration.sql [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+    "containsSeafood" BOOLEAN,
+    "containsSulfite" BOOLEAN,
+    "containsGluten" BOOLEAN,
+    "containsSesame" BOOLEAN,
+    "containsTreeNut" BOOLEAN,
+    "containsGrain" BOOLEAN,
+    "containsShellfish" BOOLEAN,
+    "containsWheat" BOOLEAN,
+    "isVegetarian" BOOLEAN NOT NULL,
+    "isVegan" BOOLEAN NOT NULL,
```
Comment (radhikadhomse) [inline backend/prisma/migrations/20250625222403_added_initial_models/migration.sql:50]: similar to the above comment, you might have more expansions on this like kosher, jain, etc

**Review by radhikadhomse — APPROVED** [review]
approving to unblock, you can make the changes in a future PR (other than the tiny comment deletion)

File: frontend/src/pages/LoginPage.tsx [hunk 14]
```diff
@@ -1,11 +1,80 @@
...
+    const { name, value } = event.target;
+    setFormData((prev) => ({ ...prev, [name]: value }));
+  };
+
+  function handleSubmit(event: React.FormEvent) {
+    event.preventDefault();
+    const valid = validateInput(formData)
+    if (valid.type === "error" && valid.text) {
+      setMessage(valid)
+      throw new Error(valid.text)
```
[the remark this answers is not in the document]
Comment (paulshaoyuqiao) [reply frontend/src/pages/LoginPage.tsx:28]: Let's actually discourage all use of `console.log` in your code if possible and `throw` statement to allow the app to keep running. Please refactor and make a new error state view component if possible.

File: frontend/src/utils/types.ts [hunk 15]
```diff
@@ -3,4 +3,19 @@ type navigationTypes = {
...
+    text: string
+}
+
+type newUserType = {
+    firebaseId: string
+    email: string
+}
+
+type formData = {
+    username: string
```
[the remark this answers is not in the document]
Comment (paulshaoyuqiao) [reply frontend/src/utils/types.ts:17]: Can we also follow a more verbose convention for type definition:

- Let's start the first letter of each word with upper case
- name should try to follow this format `Recipe<use case><UI/surface><content>` --> in your case, it would be called something like `RecipeAuthFormData`

File: backend/prisma/migrations/20250625222403_added_initial_models/migration.sql [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+    "containsSoy" BOOLEAN,
+    "containsEgg" BOOLEAN,
+    "containsSeafood" BOOLEAN,
+    "containsSulfite" BOOLEAN,
+    "containsGluten" BOOLEAN,
+    "containsSesame" BOOLEAN,
+    "containsTreeNut" BOOLEAN,
+    "containsGrain" BOOLEAN,
+    "containsShellfish" BOOLEAN,
+    "containsWheat" BOOLEAN,
```
[the remark this answers is not in the document]
Comment (paulshaoyuqiao) [reply backend/prisma/migrations/20250625222403_added_initial_models/migration.sql:48]: yeah actually I like @radhikadhomse 's suggestions better (I originally suggested booleans --> wasn't expecting so many allergens). 

@Brianna-Flo, if we want to expose all these allergens on the UI (which might also be a problem), let's actually use a list of texts/or enums (ints) and declare all allergens as enums for reusing.

You might want to think about how the UI is going to handle this to avoid multiple allergens from being overwhelming. Another idea is to potentially abstract all the specific ingredients away and create broader categories. 

For example:
* containsPeanuts
* containsTreeNut
can be just nuts (if users want, they can look for more details)
